### PR TITLE
AC_WPNav: ensure parameters meet original constraints

### PR DESCRIPTION
**Pull request by hendjoshsr71**
The parameter `wp_radius_cm` is checked in `AP_WPNav's` constructor, but the parameters get loaded after the constructor. If `WPNAV_RADIUS`  is less than `WPNAV_WP_RADIUS_MIN`, you can end up in a situation where getting to the current waypoint is never registered even if we are within `WPNAV_WP_RADIUS_MIN`.

Also can happen if you update the parameters for example using mavproxy.

To replicate set WPNAV_RADIUS to 0. Most apparent when OA is enabled or upon landing.

### CODE REVIEW

**Comment by muramura** [comment]
I wrote the following version to CUBE Orange and set the WPNAV_radius to 0.
I plugged and unplugged the power supply.
I made sure that the value stayed at 0.
I was able to confirm that the minimum value was 5 cm by this PR and that it was 5 cm by unplugging and plugging the power supply.

Verification version
ArduCopter V4.1.0-dev (a6bb9dfc)

File: libraries/AC_WPNav/AC_WPNav.cpp [hunk 1]
```diff
@@ -132,6 +132,11 @@ void AC_WPNav::wp_and_spline_init(float speed_cms)
         _wp_accel_cmss.set_and_save(WPNAV_ACCELERATION);
     }
 
+    // check _wp_radius_cm is reasonable
+    if (_wp_radius_cm < WPNAV_WP_RADIUS_MIN) {
```
Comment (muramura) [inline libraries/AC_WPNav/AC_WPNav.cpp:136]: I think the following process is a good one.
SET_AND_SAVE should be set even if there is no change, and sent by telemetry.
IFCHANGED, do not set unless the value is changed.

_wp_radius_cm.set_and_save_ifchanged(MAX(_wp_radius_cm, WPNAV_WP_RADIUS_MIN));

**Comment by muramura** [comment]
I think the constructor process will be deletion.
https://github.com/ArduPilot/ardupilot/pull/17673/files#L99

**Comment by muramura** [comment]
I think the _wp_accel_cmss process will also be deleted.

File: libraries/AC_WPNav/AC_WPNav.cpp (hunk 1 above)
Comment (hendjoshsr71) [reply libraries/AC_WPNav/AC_WPNav.cpp]: Ahh I ddin't know about that function. It saves a few bytes so I went with it. Though functionally I think both versions are equivalent and my original version saves a bit of work each loop.

File: libraries/AC_WPNav/AC_WPNav.cpp [hunk 2]
```diff
@@ -127,10 +123,14 @@ AC_WPNav::TerrainSource AC_WPNav::get_terrain_source() const
 ///     should be called once before the waypoint controller is used but does not need to be called before subsequent updates to destination
 void AC_WPNav::wp_and_spline_init(float speed_cms)
 {
+    
+    // sanity check parameters
```
Comment (rmackay9) [inline libraries/AC_WPNav/AC_WPNav.cpp:127]: Let's remove this extra comment at the top, it's pretty obvious what's going on from the other comment immediately below.

**Comment by rmackay9** [comment]
Merged, thanks!  I made a comment about a comment but then decided we need to keep moving forward so txs!

**Pull request merged**